plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -1297,6 +1297,14 @@ public ColumnHandle getMergeRowIdColumnHandle(ConnectorSession session, Connecto
     @Override
     public ConnectorMergeTableHandle beginMerge(ConnectorSession session, ConnectorTableHandle tableHandle, RetryMode retryMode)
     {
+        if (retryMode != NO_RETRIES) {
+            if (!jdbcClient.supportsMergeRetries()) {
+                throw new TrinoException(NOT_SUPPORTED, "This connector does not support MERGE with retries");
+            }
+            if (isNonTransactionalInsert(session)) {
```

Comment:
This condition was never checked, so it could be removed as the above `if` was always throwing an exception. It looks like we can simplify the code to:
```
if (retryMode != NO_RETRIES) {
  throw new TrinoException(NOT_SUPPORTED, "This connector does not support MERGE with fault-tolerant execution");
}
```